frontend/src/features/search/Search.tsx

```diff
@@ -23,57 +27,104 @@ const Search: React.FC<{}> = () => {
   const searchResults = useSelector(selectSearchResults)
   const pageInfo = useSelector(selectSearchPageInfo)
 
-  const { q, page } = NewSearchParams(queryString.parse(search))
+  const searchParams = NewSearchParams(queryString.parse(search))
+  const { q, page, sort } = searchParams
 
   // if the query string ever changes, fetch new data
   useEffect(() => {
-    dispatch(loadSearchResults(q, page))
-  }, [q, page, dispatch])
+    dispatch(loadSearchResults(searchParams))
+  }, [q, page, sort, dispatch])
 
   // handle new search term input
   const handleSearchSubmit = (newQ:string) => {
-    const newParams = new URLSearchParams(`q=${newQ}`)
-    history.push({ search: newParams.toString() })
+    updateQueryParams({ q: newQ })
   }
 
   // handle page change from PageControl
   const handlePageChange = ({ selected: pageIndex }: PageChangeObject) => {
-    const newPage = pageIndex + 1
-    const newParams = new URLSearchParams(`q=${q}&page=${newPage}`)
-    history.push({ search: newParams.toString() })
-    if (scrollContainer.current) {
+    updateQueryParams({ page: pageIndex + 1 })
+  }
+
+  // merges new query params with existing params, updates history
+  const updateQueryParams = (newQueryParams: Record<string, any>) => {
```

Comment:
dig it! like that there is only one place to look to see by what rules the params are shifting